src/app/app_execution.rs

```diff
@@ -108,3 +110,179 @@ impl AppExecution {
         Ok(())
     }
 }
+
+/// A stream of [`RecordBatch`]es that can be paginated for display in the TUI.
```

Comment:
One thing that isn't entirely clear to me is if an entire RecordBatch is shown at a time or just a slice (like only rows 100-200)

This might be an important distinction in terms of what "current batch" means and if you want this structure to pagninate based on batch index or logical row number

I think either could work -- but if you pagniate on batch, you'll have to implement logic somewhere else to translate that into logical row number for display

Reply:
I think that could depend on the UI that we expose.  I have two options in mind:

1. Incrementing a page completely replaces the displayed records with the next page of records
2. Incrementing a page _adds_ the records to the existing view (i think getting the UI correct for this may not be straight forward as i believe the naive expectation could be that the next page is triggered while scrolling and reaching the bottom of the table - like how it works in DBeaver. But happy to hear other views on this)

One thing to note - is that the `Table` widget we use automatically provides some scrolling capabilities (it tracks the selected row automatically - so if we continue to use that, which i think is beneficial at our stage, we at least dont have to do everything from scratch).

One potentially terrible idea I had, that would at least enable a _very_ simple v1 (at the cost of worse query performance) is default batch size for the TUI to some relatively small amount (say 200 rows) and then all rows for that batch would be added to the `Table`. Next page just gets the next batch and replaces the `Table` records. (I.e. option 1). We wouldnt need to track rows or figure out how to stitch records together between record batch boundaries.  I believe this is the simplest approach and would at least make the app usable for larger queries, then we could add a todo for something more user friendly / that doesnt impact query performance.

Reply:
> Incrementing a page completely replaces the displayed records with the next page of records

This is how I (naively) as a user would expect things to behave

Reply:
> One potentially terrible idea I had, that would at least enable a very simple v1 (at the cost of worse query performance) is default batch size for the TUI to some relatively small amount (say 200 rows) and then all rows for that batch would be added to the Table. Next page just gets the next batch and replaces the Table records. (I.e. option 1). We wouldnt need to track rows or figure out how to stitch records together between record batch boundaries. I believe this is the simplest approach and would at least make the app usable for larger queries, then we could add a todo for something more user friendly / that doesnt impact query performance.

This seems reasonable to me. 

I also think all the APIs to stitch rows together are in arrow-rs (like `concat_batches` and slice) so we could also make some sort of adapter stream (another wrapper!) that took the incoming stream and reformatted it to smaller sized record batches